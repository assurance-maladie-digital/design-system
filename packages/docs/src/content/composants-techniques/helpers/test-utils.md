---
title: testUtils
description: Le helper `testUtils` contient un ensemble de fonctions utilitaires pour les tests unitaires.
---

<doc-tabs>

<doc-tab-item label="Utilisation">

<doc-alert type="info">

Le helper `testUtils` exporte également toutes les fonctions du package [Vue Test Utils](https://vue-test-utils.vuejs.org/).

</doc-alert>

### applyGlobalConfig

La fonction `applyGlobalConfig` permet d’appliquer la configuration globale pour les tests, elle est utilisée dans le fichier `setup.ts`.

```ts
import { applyGlobalConfig } from '@cnamts/vue-dot/src/helpers/testUtils';

applyGlobalConfig();
```

### installGlobalPlugins

La fonction `installGlobalPlugins` permet d’installer les plugins [Vue Input Facade](https://ronaldjerez.github.io/vue-input-facade/), [Vue Meta](https://vue-meta.nuxtjs.org/), [Vuex](https://vuex.vuejs.org/), [Vuetify](https://v2.vuetifyjs.com/) et Vue Dot pour les tests.

```ts
import { createLocalVue, installGlobalPlugins } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installGlobalPlugins(localVue);
```

### installRouter

La fonction `installRouter` permet d’installer [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createLocalVue, installRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installRouter(localVue);
```

### createRouter

La fonction `createRouter` permet de créer une instance de [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const routes = [];

const router = createRouter({
	routes
});
```

### createStore

La fonction `createStore` permet de créer une instance de [Vuex](https://vuex.vuejs.org/) pour les tests.

```ts
import { createStore } from '@cnamts/vue-dot/src/helpers/testUtils';

const store = createStore({
	state: {}
});
```

### createVuetifyInstance

La fonction `createVuetifyInstance` permet de créer une instance de [Vuetify](https://v2.vuetifyjs.com/) pour les tests.

```ts
import { createVuetifyInstance } from '@cnamts/vue-dot/src/helpers/testUtils';

const vuetify = createVuetifyInstance();
```

### mockVFormRef

La fonction `mockVFormRef` permet de mocker la référence d’un composant [VForm](https://v2.vuetifyjs.com/en/components/forms/) pour les tests.

```ts
import { mockVFormRef } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	$refs: {
		form: mockVFormRef(true)
	}
};
```

### mockTranslations

La fonction `mockTranslations` permet de mocker les traductions pour les tests.

```ts
import { mockTranslations } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	...mockTranslations<string | string[]>({
		'path.to.title': 'Titre',
		'path.to.items': [
			'Élément 1',
			'Élément 2',
			'Élément 3'
		]
	})
};
```

</doc-tab-item>

<doc-tab-item label="API">
<doc-api name="helpers/test-utils"></doc-api>
</doc-tab-item>

</doc-tabs>
