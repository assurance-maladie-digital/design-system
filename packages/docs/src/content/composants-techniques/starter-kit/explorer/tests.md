Le dossier `tests` contient les tests d’interface et les tests unitaires globaux de l’application.

<doc-alert type="warning">
Il n’est pas recommandé de renommer ce dossier.
</doc-alert>
