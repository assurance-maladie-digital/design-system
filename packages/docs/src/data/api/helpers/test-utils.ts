import { Api } from '~/types';

export const api: Api = {
	applyGlobalConfig: {
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installGlobalPlugins: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installRouter: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	createRouter: {
		arguments: [
			{
				name: 'options',
				description: 'Router options',
				type: 'RouterOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'VueRouter',
			},
		],
	},
	createStore: {
		arguments: [
			{
				name: 'options',
				description: 'Store options',
				type: 'StoreOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'Store',
			},
		],
	},
	createVuetifyInstance: {
		arguments: [
			{
				name: 'preset',
				description: 'Vuetify preset',
				type: 'Partial<UserVuetifyPreset>',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'Vuetify',
			},
		],
	},
	mockVFormRef: {
		arguments: [
			{
				name: 'isValid',
				description: 'Is valid',
				type: 'boolean',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	mockTranslations: {
		arguments: [
			{
				name: 'translations',
				description: 'Translations',
				type: 'IndexedObject',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
};
