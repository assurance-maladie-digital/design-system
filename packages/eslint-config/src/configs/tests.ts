import { type FlatESLintConfigItem } from 'eslint-define-config';
import { pluginVitest } from '../plugins';
import { GLOB_TESTS } from '../globs';
import { isInEditor } from '../env';

export const tests: FlatESLintConfigItem[] = [
	{
		plugins: {
			tests: pluginVitest,
		},
	},
	{
		files: GLOB_TESTS,
		rules: {
			'tests/consistent-test-it': [
				'error',
				{
					fn: 'it',
					withinDescribe: 'it',
				},
			],
			'tests/expect-expect': 'error',
			'tests/no-alias-methods': 'error',
			'tests/no-commented-out-tests': isInEditor ? 'off' : 'error',
			'tests/no-conditional-expect': 'error',
			'tests/no-disabled-tests': 'error',
			'tests/no-done-callback': 'error',
			'tests/no-duplicate-hooks': 'error',
			'tests/no-focused-tests': isInEditor ? 'off' : 'error',
			'tests/no-identical-title': 'error',
			'tests/no-interpolation-in-snapshots': 'error',
			'tests/no-standalone-expect': 'error',
			'tests/no-test-return-statement': 'error',
			'tests/prefer-comparison-matcher': 'error',
			'tests/prefer-equality-matcher': 'error',
			'tests/prefer-hooks-in-order': 'error',
			'tests/prefer-hooks-on-top': 'error',
			'tests/prefer-mock-promise-shorthand': 'error',
			'tests/prefer-spy-on': 'error',
			'tests/prefer-to-be-falsy': 'error',
			'tests/prefer-to-be-object': 'error',
			'tests/prefer-to-be-truthy': 'error',
			'tests/prefer-to-be': 'error',
			'tests/prefer-to-contain': 'error',
			'tests/prefer-to-have-length': 'error',
			'tests/prefer-todo': 'error',
			'tests/require-hook': 'error',
			'tests/require-top-level-describe': 'error',
			'tests/valid-describe-callback': 'error',
			'tests/valid-expect': 'error',
			'tests/valid-title': 'error',
		},
	},
];
