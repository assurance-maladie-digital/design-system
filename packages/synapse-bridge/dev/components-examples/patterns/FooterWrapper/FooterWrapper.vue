<script lang="ts" setup>
import { FooterWrapper } from '@/main'
import FooterBtn from '@/patterns/FooterWrapper/FooterBtn'
</script>

<template>
	<FooterWrapper>
		<FooterBtn>
			Mentions l√©gales
		</FooterBtn>

		<FooterBtn>
			CGU
		</FooterBtn>

		<FooterBtn inert>
			Version 1.0.0
		</FooterBtn>
	</FooterWrapper>

	<FooterWrapper>
		<FooterBtn>
			CGU
		</FooterBtn>

		<FooterBtn inert>
			Version 1
		</FooterBtn>

		<FooterBtn inert>
			28/05/2020
		</FooterBtn>
	</FooterWrapper>

	<FooterWrapper>
		<VRow>
			<VSpacer />

			<p class="mb-0">
				Version 1
			</p>
		</VRow>
	</FooterWrapper>
</template>

<style lang="scss" scoped>
.v-layout + .v-layout {
	margin-top: 60px;
}
</style>
