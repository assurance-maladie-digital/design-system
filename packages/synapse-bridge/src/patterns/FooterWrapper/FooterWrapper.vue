<script lang="ts">
	import { defineComponent } from 'vue'

	import consola from 'consola'

	export default defineComponent({
		mounted() {
			consola.warn(
				'FooterWrapper is deprecated since v2.2.0, use FooterBar instead.'
			)
		},
	})
</script>

<template>
	<VLayout>
		<VFooter
			width="100%"
			elevation="1"
			color="#fafafa"
			class="vd-footer-wrapper"
		>
			<VRow
				justify="center"
				class="ma-0"
			>
				<slot />
			</VRow>
		</VFooter>
	</VLayout>
</template>

<style lang="scss" scoped>
	.v-footer {
		min-height: 64px;
	}
</style>
