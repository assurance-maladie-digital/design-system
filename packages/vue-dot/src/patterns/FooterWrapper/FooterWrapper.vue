<template>
	<VFooter
		min-height="64"
		width="100%"
		elevation="1"
		color="#fafafa"
		class="vd-footer-wrapper"
	>
		<VRow
			class="ma-0"
			justify="center"
		>
			<slot />
		</VRow>
	</VFooter>
</template>

<script lang="ts">
	import Vue from 'vue';
	import Component from 'vue-class-component';

	import consola from 'consola';

	/** @deprecated Use FooterBar instead */
	@Component
	export default class FooterWrapper extends Vue {
		mounted() {
			// eslint-disable-next-line no-console
			consola.warn('FooterWrapper is deprecated since v2.2.0, use FooterBar instead.');
		}
	}
</script>
