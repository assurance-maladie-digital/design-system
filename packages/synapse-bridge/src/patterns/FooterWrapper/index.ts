import FooterWrapper from './FooterWrapper.vue'

export default FooterWrapper
