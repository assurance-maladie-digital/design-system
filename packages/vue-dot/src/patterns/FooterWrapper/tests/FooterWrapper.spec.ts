import Vue from 'vue';
import { Wrapper } from '@vue/test-utils';

import consola from 'consola';

import { mountComponent } from '@/tests';

import FooterWrapper from '../';

let wrapper: Wrapper<Vue>;

describe('FooterWrapper', () => {
	const spy = jest.fn();

	beforeEach(() => {
		consola.mockTypes(() => spy);
	});

	it('renders correctly', () => {
		wrapper = mountComponent(FooterWrapper);

		expect(wrapper).toMatchSnapshot();
		expect(spy).toHaveBeenCalledWith('FooterWrapper is deprecated since v2.2.0, use FooterBar instead.');
	});
});
