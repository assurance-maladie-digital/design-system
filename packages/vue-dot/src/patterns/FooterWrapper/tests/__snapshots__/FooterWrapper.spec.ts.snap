// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterWrapper renders correctly 1`] = `
<vfooter-stub color="#fafafa" elevation="1" height="auto" minheight="64" width="100%" tag="footer" class="vd-footer-wrapper">
  <vrow-stub tag="div" justify="center" class="ma-0"></vrow-stub>
</vfooter-stub>
`;
